### ISSUE TITLE
FP S6966: ServerConnection.ExecuteNonQuery (SMO) has no async counterpart

### ISSUE THREAD

**Issue by tsimbalar** [SonarSource/sonar-dotnet#9802]
## Description

Rule S6966 flags `ServerConnection.ExecuteNonQuery()` from Microsoft SQL Server Management Objects (SMO) with:

> Await ExecuteNonQueryAsync instead.

However, [`ServerConnection.ExecuteNonQuery`](https://learn.microsoft.com/en-us/dotnet/api/microsoft.sqlserver.management.common.serverconnection.executenonquery?view=sql-smo-160) has **no async counterpart**. The `ServerConnection` class (from `Microsoft.SqlServer.SqlManagementObjects`) does not provide an `ExecuteNonQueryAsync` method.

The analyzer likely matches the method name against `SqlCommand.ExecuteNonQueryAsync` from a different class hierarchy (`System.Data.SqlClient` / `Microsoft.Data.SqlClient`), but these are unrelated types.

## Repro

```csharp
using Microsoft.Data.SqlClient;
using Microsoft.SqlServer.Management.Common;
using Microsoft.SqlServer.Management.Smo;

async Task RunSqlScript(string connectionString, string sqlScript)
{
    await using var connection = new SqlConnection(connectionString);
    await connection.OpenAsync();
    var server = new Server(new ServerConnection(connection));
    server.ConnectionContext.ExecuteNonQuery(sqlScript); // S6966: Await ExecuteNonQueryAsync instead.
}
```

## Expected behavior

No S6966 warning, since `ServerConnection` does not offer an async alternative.

## Environment

- SonarAnalyzer.CSharp: **10.23.0.137933**
- Microsoft.SqlServer.SqlManagementObjects: **181.19.0**
- .NET 9.0

**Comment by tsimbalar** [comment]
Closing to re-create with the correct issue template.
